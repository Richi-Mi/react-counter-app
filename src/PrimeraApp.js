// import React, { Fragment } from 'react';
// return (
//     <Fragment>
//         <h1 className='title'> saludo </h1>
//         <p> Mi primera Apilcación </p>    
//     </Fragment>
// );
import React from 'react';
import PropTypes from 'prop-types';

// FC

const PrimeraApp = ({ saludo, edad, subtitulo = '' }) => {
    // if( !saludo ) {
    //     throw new Error('El saludo es Necesario')
    // }

    return ( 
        < >
            <h1> { saludo } </h1>
            {/* <pre > { JSON.stringify(saludo, null, 3)} </pre> */}
            <h2>{ subtitulo }</h2>
            <p> Mi primera Apilcación, edad: { edad } </p>    
        </ >
    )
}
PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired,
    edad: PropTypes.number
}
PrimeraApp.defaultProps = {
    subtitulo: 'Soy un subtitulo'
}
export default PrimeraApp;