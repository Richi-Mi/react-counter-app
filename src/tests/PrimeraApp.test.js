import React from 'react';
// import { render } from "@testing-library/react";
import PrimeraApp from "../PrimeraApp";
import { shallow } from 'enzyme';

/**
 * Renderizar. es un anglicismo para representación gráfica, usado en la jerga informática para referirse 
 * al proceso de generar imagen fotorrealista, o no, a partir de un modelo 2D o 3D.
 * 
 * Podemos hacer pruebas de componentes de React usando la extensión de Jest, o la libreria Enzyme, pero esta solo esta
 * disponible en la version 16 de React, por ahora
 * */ 

describe('Pruebas en <PrimeraApp />', () => {

    // Forma de hacerlo con Jest
    // test('Mostrar el mensaje Hola Papu', () => {
    //     const saludo = "Hola Papu";
    //     /// render( ReactComponent ) .- Renderiza un componente de React, y lo guarda en wrapper
    //     const wrapper = render( <PrimeraApp saludo={ saludo } /> )
    //     const isText = wrapper.getByText( saludo )

    //     // Para usar la función .toBeInTheDocument() es necesario crear el archivo setUpTests.js y 
    //     // ahi hacer la importación necesaria
    //     expect( isText ).toBeInTheDocument()
    // })

    // Forma de Hacerlo con Enzyme.
    test('Mostrar el mensaje, Hola Papu - Enzime', () => {
        const saludo = 'Hola Papu'
        const wrapper = shallow( <PrimeraApp saludo={ saludo }/>)

        // .toMatchSnapshot() .- Validamos que la snapshot( codigo html ) de el componente sea el mismo que el wrapper( codigo html)
        expect( wrapper ).toMatchSnapshot();
    })
    
    test('Debe mostrar el subtitulo enviado por props', () => {
        const saludo = 'Hola Papu'
        const subTitle = 'Hola Soy un subtitulo'
        const wrapper = shallow( <PrimeraApp saludo={ saludo } subtitulo={ subTitle } edad={15}/>)

        const textoParrafo = wrapper.find('h2').text() // Devuelve el texto del h2
    
        // const textoParrafo = wrapper.find('h1') .- Devuelve un elemento del componente por su nombre
        // const textoParrafo = wrapper.find('#txt_name') .- Devuelve un elemento del componente por su id
        // const textoParrafo = wrapper.find('.elements') .- Devuelve un elemento del componente por su clase

        expect( textoParrafo ).toBe( subTitle )
    })
    
})
