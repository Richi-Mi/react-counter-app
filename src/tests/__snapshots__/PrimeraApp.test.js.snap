// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> Mostrar el mensaje, Hola Papu - Enzime 1`] = `
<Fragment>
  <h1>
     
    Hola Papu
     
  </h1>
  <h2>
    Soy un subtitulo
  </h2>
  <p>
     Mi primera Apilcaci√≥n, edad: 
     
  </p>
</Fragment>
`;
